docs/DeveloperGuide.md

```diff
@@ -4,26 +4,207 @@
+------------|-------------------|----------------------|----------------------------|        
+add | `"add_user ..."`  | `"add_lesson ..."`   | `"add_meeting ..."`     
+AddCommand | `AddUserCommand`  | `AddLessonCommand`   | `AddMeetingCommand`  
+
+
+## 3.1.1 Add User Feature
+`add_user` adds new user and his or her timetable into the master timetable.
+
+The following sequence diagram shows how the `add_user` command works:
+![AddUserCommandSequenceDiagram](diagrams/AddUserCommandSequenceDiagram.png)
+
+
+## 3.1.2 Add Events Feature
+
+`add_lesson` allows users to create a new lesson and add it to their own timetables.
+
+`add_meeting` allows users to create a new meeting and add it to all the existing timetables.
+
+
+The following sequence diagram shows how the `add_lesson` operation works in detail.
+![AddMeetingCommandSequenceDiagram](diagrams/AddLessonCommandSequenceDiagram.png)
+
+Given below is the steps on how the `add_lesson` operation works.
+
+>To be added
+
+The following sequence diagram shows how the `add_meeting` operation works in detail.
+![AddMeetingCommandSequence](diagrams/AddMeetingCommandSequence.png)
+
+Given below is the steps on how the `add_meeting` operation works.
+
+>To be added
+> 
+## 3.2 Listing Events Feature
+The `list` command is a command that the user can input in order to list out the events he has in his timetable.
+
+`list [user]` displays the timetable for the particular user.
+
+`list all` displays the timetable for all users.
+
+In essence, the `list all` command repeatedly calls `list [user]` for all users.
+
+Before the timetable is listed out, it will also be sorted according to day and time for easy reading.
+
+The following sequence diagram shows how the command `` is executed.
+![ListCommandSequenceDiagram](diagrams/ListCommandSequenceDiagram.png)
+
+## 3.3 Finding Common Free Timeslots Feature
+The `free` command is a command that the user can input in order to find timeslots where all users are free.
+
+For greater customisation, `free [duration]` displays all common timeslots which has a duration longer than specified.
+
+The searching algorithm works by marking the busy slots of each timetable first.
+Timings that are not marked 'busy' are then identified as free time slots.
+
+The following sequence diagram shows how the command `free` is executed.
+![FreeCommandSequenceDiagram](diagrams/FreeCommandSequenceDiagram.png)
+
+## 3.4 Delete events `delete`
+Deletes an event from the user's specified timetable
+
+**Format:** ` delete n/NAME i/INDEX`
+
+* Deletes from the timetable of user
+* Deletes the lesson at the specified `INDEX`.
+* The `INDEX` refers to the index number shown in the displayed lesson list.
+
+Example of usage:
+
+`delete n/John i/1`
+
+If the event is a meeting, it will delete this event from all users
+
+The following sequence diagram shows how the `delete` command works:
+![DeleteCommandSequenceDiagram](diagrams/DeleteCommandSequenceDiagram.png)
```

Comment:
How does the control flow go from 1 to 2? Or should the activation bar end at 1?
<img width="525" alt="image" src="https://user-images.githubusercontent.com/91527581/161183702-d32e28ac-b883-46c2-96c3-9b9ae78d3634.png">